### DISCUSSION TITLE
x-on:click triggers twice if it includes ID

### DISCUSSION THREAD

**Discussion by kesirog** [alpinejs/alpine#2291] [7. Bugs (archived)]
```
<div class="parent" x-data="{}">
    <div class="child" id="abc" x-on:click="console.log('abc')">abc</div>
</div>
```
Because of the **id** on the div or any html element, click will trigger twice. This doesn't look intended to me, if it is, please let me know how to tackle this.

**Comment by SimoTod** [comment]
Why do you say that? 
P.s. Your example doesn't fire twice.

**Reply by kesirog** [reply]
Sorry for the delay, plus I've forgot that I already opened a discussion on this topic, please close the new one, we can continue here.

**Component use example for surface:**
```
<SomeSurfaceComponent name={:something} id={"something-#{@changeset.data.id}"}/>
```
**Component itself:**
```
defmodule Project.Components.SomeSurfaceComponent do
  use Project, :surface_component

  ....some surface props and data...
  prop id, :string

  def render(assigns) do
    ~F"""
      <Context get={Surface.Components.Form, form: form}>
        <div x-data="someFunction()" {=@id}>
          <template x-for="(item, index) in items()" :key="index">
            <div x-text="item" x-on:click="doSomething()">
          </template>
        </div>
      </Context>
    """
  end
end
```

We also keep JS functions in separate files.

```
{=@id} translates to id="variable"
```

If we keep x-data and native html id attribute on the same element, x-data and all nested clicks duplicate and template will output double amount of content, inits are called twice etc etc. As it seems there might be more people facing this with surface, since I've noticed a few of them on stackoverflow.

My workaround is to remove **id**  from our component on both sides, then every thing works as intended. Is this core mechanic or it's know bug/issue? And what's the solution? This bug is present for at least half a year.

I can also make a loom video if it would help.

**Reply by ekwoka** [reply]
Trying to recreate this minimally somewhere would be helpful.

Your description seems to be implying it's initializing the tree on the element twice, but it seems more like it's a conflict with your system elsewhere but hard to tell from just what you've shown.
